src/middlewared/middlewared/async_validators.py

```diff
@@ -22,27 +21,32 @@ async def check_path_resides_within_volume(verrors, middleware, name, path, glus
     ):
         verrors.add(name, "The path must reside within a pool mount point")
 
-    if osc.IS_LINUX:
-        # we must also make sure that any sharing service does not point to
-        # anywhere within the ".glusterfs" dataset since the clients need
-        # to go through the appropriate gluster client to write to the cluster.
-        # If data is modified directly on the gluster resources, then it will
-        # cause a split-brain scenario which means the data that was modified
-        # would not be sync'ed with other nodes in the cluster.
-        rp = Path(rp)
-
-        using_gluster_path = False
-        if rp.is_mount() and rp.name == '.glusterfs':
-            using_gluster_path = True
-        else:
-            # subtract 2 here to remove the '/' and 'mnt' parents
-            for i in range(0, len(rp.parents) - 2):
-                if rp.parents[i].is_mount() and rp.parents[i].name == ".glusterfs":
-                    using_gluster_path = True
-                    break
-
-        if using_gluster_path:
-            verrors.add(name, "A path being used by Gluster is not allowed")
+    # we must also make sure that any sharing service does not point to
+    # anywhere within the ".glusterfs" dataset since the clients need
+    # to go through the appropriate gluster client to write to the cluster.
+    # If data is modified directly on the gluster resources, then it will
+    # cause a split-brain scenario which means the data that was modified
+    # would not be sync'ed with other nodes in the cluster.
+    rp = Path(rp)
+
+    using_gluster_path = False
+    if rp.is_mount() and rp.name == '.glusterfs':
+        using_gluster_path = True
+    else:
+        # subtract 2 here to remove the '/' and 'mnt' parents
+        for i in range(0, len(rp.parents) - 2):
+            if rp.parents[i].is_mount() and rp.parents[i].name == ".glusterfs":
+                using_gluster_path = True
+                break
+
+    if using_gluster_path:
+        verrors.add(name, "A path being used by Gluster is not allowed")
+
+    k8s_ds = (await middleware.call('kubernetes.config'))['dataset']
```

Comment:
Could use a little note here as well, to sync with above note about cluster:
```suggestion

    # We must make sure that the internal Apps datasets are not shared, 
    # as this might lead to breaking changes to its, very specific, structure and settings.
    k8s_ds = (await middleware.call('kubernetes.config'))['dataset']
```